db/db_iter.cc

```diff
@@ -1234,23 +1236,38 @@ bool DBIter::TooManyInternalKeysSkipped(bool increment) {
 }
 
 bool DBIter::IsVisible(SequenceNumber sequence) {
-  return sequence <= sequence_ &&
+  return sequence <= MaxVisibleSequenceNumber() &&
          (read_callback_ == nullptr || read_callback_->IsCommitted(sequence));
```

Comment:
We have changed the semantics of IsCommitted by returning true for our-own-writes. Then IsCommitted is not the right name. We should either rename it or add this check for our-own-write to a new function.